packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -131,6 +142,27 @@ const ProgressMessage: { [key: string]: string } = Object.freeze({
   [DepsType.FuncCoreTools]: `Checking and installing ${DepsDisplayName[DepsType.FuncCoreTools]}`,
 });
 
+type PortCheckerInfo = { checker: Checker.Ports; ports: number[] };
+type PrerequisiteCheckerInfo = { checker: Checker | DepsType } | PortCheckerInfo;
```

Comment:
`PrerequisiteCheckerInfo` can be cleaner without `| PortCheckerInfo`.
Just convert to `PortCheckerInfo` for port checker. `(info as PortCheckerInfo).ports...`